lms/services/dashboard.py

```diff
@@ -47,22 +48,39 @@ def get_request_course(self, request):
 
         return course
 
-    def get_request_organization(self, request):
-        """Get and authorize an organization for the given request."""
-        organization = self._organization_service.get_by_public_id(
-            public_id=request.matchdict["organization_public_id"]
-        )
-        if not organization:
-            raise HTTPNotFound()
+    def get_request_organizations(self, request) -> list[Organization]:
+        """Get the relevant organizations for the current requests."""
+        organizations = []
 
-        if request.has_permission(Permissions.STAFF):
-            # STAFF members in our admin pages can access all organizations
-            return organization
+        # If we have an user, include the organization it belongs to
+        if request.lti_user:
+            lti_user_organization = request.lti_user.application_instance.organization
+
+            if not self._organization_service.is_member(
+                lti_user_organization, request.user
+            ):
+                raise HTTPUnauthorized()
 
-        if not self._organization_service.is_member(organization, request.user):
+            organizations.append(lti_user_organization)
+
+        # Include any other organizations we are an admin in
```

Comment:
This serves school admins and, for now, also staff members, ie you will need to add yourself as an organization admin to see its courses.